sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroSource.java

```diff
@@ -393,7 +393,7 @@ private static InputStream decodeAsInputStream(byte[] data, String codec) throws
       ByteArrayInputStream byteStream = new ByteArrayInputStream(data);
       switch (codec) {
         case DataFileConstants.SNAPPY_CODEC:
-          return new SnappyCompressorInputStream(byteStream);
+          return new SnappyCompressorInputStream(byteStream, 1 << 16 /* Avro uses 64KB blocks */);
```

Comment:
That's all fair. Looking deeper, it looks like the change is fairly well-scoped and also allows a cleaner set of Maven dependencies. (As I'm sure you already realized).